tee-worker/omni-executor/rpc-server/src/methods/omni/submit_user_op.rs

```diff
@@ -0,0 +1,192 @@
+			address.copy_from_slice(&address_bytes);
+
+			// Collect unique sender addresses to avoid redundant validation calls
+			let unique_addresses: HashSet<Address> = params
+				.user_operations
+				.iter()
+				.map(|op| {
+					op.sender.parse::<Address>().map_err(|e| {
+						error!("Invalid sender address '{}': {}", op.sender, e);
+						PumpxRpcError::from_error_code(ErrorCode::ParseError)
+					})
+				})
+				.collect::<Result<HashSet<_>, _>>()?;
+
+			// Validate each unique address once
+			for sender_address in unique_addresses {
+				if let Err(e) = validate_user_operation_ownership(
+					&user,
+					&sender_address,
+					&params.chain_id,
+					&ctx,
+				)
+				.await
+				{
+					error!(
+						"User operation ownership validation failed for sender {:?}",
+						sender_address
+					);
+					return Err(e);
+				}
+			}
+
+			let wrapper = NativeTaskWrapper::new(
+				NativeTask::SubmitUserOp(
+					AccountId::decode(&mut &address[..]).map_err(|_| {
+						error!("Failed to decode AccountId from bytes");
+						PumpxRpcError::from_error_code(ErrorCode::InternalError)
+					})?,
+					params.user_operations.clone(),
+					params.chain_id.clone(),
+					params.wallet_index,
+				),
+				None,
+				None,
+				user.client_id,
+			);
+
+			handle_omni_native_task(&ctx, wrapper, |task_ok| match task_ok {
+				NativeTaskOk::SubmitUserOp(transaction_hash) => {
+					Ok(SubmitUserOpResponse { transaction_hash })
+				},
+				_ => {
+					error!("Unexpected response type");
+					Err(PumpxRpcError::from_error_code(ErrorCode::InternalError))
+				},
+			})
+			.await
+		})
+		.expect("Failed to register omni_submitUserOp method");
+}
+
+async fn validate_user_operation_ownership(
+	user: &crate::methods::omni::common::User,
+	sender_address: &Address,
+	chain_id: &ChainId,
+	ctx: &RpcContext,
+) -> Result<(), PumpxRpcError> {
+	// Calculate the expected OA from the user's identity
+	let expected_oa_bytes = hex::decode(
+		user.omni_account.strip_prefix("0x").unwrap_or(&user.omni_account),
+	)
+	.map_err(|_| {
+		error!("Failed to decode omni account hex string");
+		PumpxRpcError::from_error_code(ErrorCode::InternalError)
+	})?;
+
+	if expected_oa_bytes.len() != 32 {
+		error!("Invalid omni account length: expected 32 bytes, got {}", expected_oa_bytes.len());
+		return Err(PumpxRpcError::from_error_code(ErrorCode::InternalError));
+	}
+
+	let expected_oa = FixedBytes::<32>::from_slice(&expected_oa_bytes);
+
+	// Get the RPC client for the chain
+	let Some(rpc_client) = ctx.rpc_clients.get(chain_id) else {
+		error!("No RPC client found for chain ID: {}", chain_id);
+		return Err(PumpxRpcError::from_error_code(ErrorCode::InternalError));
+	};
+
+	// Create a new client with the specific wallet address
+	let omni_client =
+		aa_contracts_client::OmniAccountClient::new(*sender_address, rpc_client.clone());
+
+	// Query the OmniWallet contract directly to get its stored OA
+	let stored_oa = match omni_client.get_owner().await {
```

Comment:
Why do we need to query the contract? can we just compare `user.omni_account` with `op.sender` ?

Reply:
I think we should do the check, my question is if we can compare both here instead of making the rpc call? or `sender` is not the same as `owner` ?  

Also, should we check that the user is root signer instead of owner?  (Sorry, I'm just a bit confused about how everything should work)